### PR TITLE
Fixes conservation error from removal of thin ice by adjust_enthalpy

### PR DESCRIPTION
**Pull request by akturner**
Fixes conservation error from removal of thin ice in thickness_changes in the column package by adjust_enthalpy

Fixes https://github.com/E3SM-Project/E3SM/issues/5124

[Non-BFB]

### CODE REVIEW

**Review by eclare108213 — APPROVED** [review]
Looks good, thank you.

**Comment by rljacob** [comment]
Is this non-BFB even for F-cases that use the thermo-only MPAS-seaice?

**Comment by mkstratos** [comment]
This doesn't seem to fix the issue in the F2010 compset test with dtime=1, but the previous fix did.

My test is here (I think I set it up correctly): `/lcrc/group/e3sm/ac.mkelleher/scratch/chrys/TSC.ne4_oQU240.F2010.chrysalis_intel.G.20220927_085853_dh4wu4`

**Comment by jonbob** [comment]
@mkstratos -- I can confirm that our simplified test also fails.

**Comment by akturner** [comment]
@mkstratos :Apologies, should now work

**Comment by jonbob** [comment]
Our simplified test did pass with your latest commit, @akturner -- thanks! @mkstratos, could you please test as well whenever you get a chance?

**Review by mkstratos — APPROVED** [review]
Looks good to me, the TSC baseline is generated for F2010, thanks @akturner!

**Comment by njeffery** [comment]
@akturner : I'm wondering about the salt flux to the ocean.  There is no corresponding mass flux and the salt flux is only defined based on a reference salinity.

**Comment by akturner** [comment]
@njeffery: I removed the conservation correction for salt.

**Review by njeffery — APPROVED** [review]

**Comment by jonbob** [comment]
Because this is non-BFB, I ran two ten-year tests -- one without and one with this PR. MPAS-Analysis comparison output is at:
[20221017.PR5207.anvil](https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.jwolfe/20221017.PR5207.anvil/html/)

@xylar or @mark-petersen -- could you please check this from the ocean perspective? And @eclare108213 or @njeffery -- could you please make sure the seaice looks like it's not significantly impacted?

**Comment by njeffery** [comment]
@jonbob : the sea ice looks a fair bit thicker. Are the conservation analysis members enabled in your run?  Probably worth double checking energy conservation.  Could point me to your run directory?

**Comment by njeffery** [comment]
found it!  /lcrc/group/acme/ac.jwolfe/scratch/anvil/20221017.PR5207.anvil

**Review by eclare108213 — COMMENTED** [review]
I don't really understand why the salt flux should not be updated along with the energy flux.  The current modifications fix the problem with F-cases crashing, which is great.  This change also affects the full model runs and is an improvement in terms of energy conservation.  I'm wondering whether the salt flux should also be updated. This could be done in a separate PR, to keep this one moving. On the other hand, the ice volume changes in @jonbob's tests are larger than I would expect.  Could the mismatch between salt and heat fluxes be accentuating these differences?

File: components/mpas-seaice/src/column/ice_therm_vertical.F90 [hunk 1]
```diff
@@ -1034,7 +1035,8 @@ subroutine thickness_changes (nilyr,     nslyr,    &
                                     flatn,     fsurfn,   &
                                     fcondtopn, fcondbot, &
                                     fsnow,     hsn_new,  &
-                                    fhocnn,    evapn,    &
+                                    fhocnn,    fsaltn,   &
```
[the remark this answers is not in the document]
Comment (njeffery) [reply components/mpas-seaice/src/column/ice_therm_vertical.F90:1038]: Part of the problem is that only the enthalpy of the thin ice is removed here, not it's volume.   If the volume were removed too, then both a salt and fresh water flux could be added consistently.

**Comment by jonbob** [comment]
Sorry @njeffery -- I bet I didn't. I'm happy to rerun both with the conservation analysis members on

**Review by eclare108213 — CHANGES_REQUESTED** [review]
The simulation changes are more substantial than expected, which I think should be understood before this is merged.  In my opinion, the mass (salt and fresh water) fluxes should be updated consistently with the enthalpy adjustment.  If the simulation still changes a lot, then look at conservation diagnostics to track it down.

**Pull request closed without merging**